packages/expo-modules-core/ios/FileSystemUtilities/FileSystemUtilities.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+public enum FileSystemPermissionFlags {
+  case none
+  case read
+  case write
+}
+
+public struct FileSystemUtilities {
+  @discardableResult
+  public static func ensureDirExists(at url: URL?) -> Bool {
+    guard let url else {
+      return false
+    }
+    let exists = FileManager.default.fileExists(atPath: url.path)
+
+    if !exists {
+      do {
+        try FileManager.default.createDirectory(atPath: url.path, withIntermediateDirectories: true)
+      } catch {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  public static func generatePathInCache(_ appContext: AppContext?, in directory: String, ext: String) -> String {
+    guard let appContext, let dirPath = appContext.config.cacheDirectory?.appendingPathComponent(directory) else {
+      return ""
+    }
+    let fileName = UUID().uuidString.appending(ext)
```

Comment:
```suggestion
    let fileName = UUID().uuidString.appending(extension)
```

Reply:
`extension` is a keyword so we'd have to use it like this  
```swift
let fileName = UUID().uuidString.appending(`extension`)
```
which kinda sucks 😅 Guess it's ok because we only do it once

Reply:
In such cases you can provide a custom internal name for this argument, `extension fileExtension: String`, but I think that's fine to use backticks